central/apitoken/datastore/telemetry_postgres_test.go

```diff
@@ -0,0 +1,125 @@
+//go:build sql_integration
+
+package datastore
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	pgStore "github.com/stackrox/rox/central/apitoken/datastore/internal/store/postgres"
+	"github.com/stackrox/rox/central/apitoken/testutils"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/postgres/pgtest"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/sac/resources"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGather(t *testing.T) {
+	pool := pgtest.ForT(t)
```

Comment:
if we move setup/cleanup to test cases we can run them in parallel and we won't need `Delete` 
```suggestion
	pool := pgtest.ForT(t)
	t.Cleanup(func() {
		testingDB.Teardown(t)
	})
```

Reply:
I found that `ForT` has a problem with long test names. Maybe we can change it to hash of name
```
// CreateADatabaseForT creates a postgres database for test
func CreateADatabaseForT(t testing.TB) string {
	suffix, err := random.GenerateString(5, random.AlphanumericCharacters)
	require.NoError(t, err)

	h := fnv.New64a()
	_, err = io.WriteString(h, t.Name())
	require.NoError(t, err)

	database := fmt.Sprintf("%x_%s", h.Sum64(), suffix)

	CreateDatabase(t, database)

	return database
}
```

Reply:
Isn't this much more complicated than a simple delete function?